go/chat/boxer.go

```diff
@@ -219,6 +222,10 @@ func (b *Boxer) boxMessageV1(ctx context.Context, msg chat1.MessagePlaintextV1,
 		}
 	}
 
+	if len(msg.ClientHeader.TlfName) == 0 {
```

Comment:
Yes, we try to do so everywhere.  There is a small window in which the server is still running and the client is new, so even in that case.

The key idea here is never to change the type of a parameter.  If needs be, you can introduce a new RPC and mark the previous  one for deprecation.